swift/apple/FirezoneKit/Sources/FirezoneKit/Stores/AuthStore.swift

```diff
@@ -230,9 +237,22 @@ public final class AuthStore: ObservableObject {
       Task {
         await self.signOut()
       }
+      #if os(macOS)
+        self.showSignedOutAlertmacOS()
+      #endif
     }
   }
 
+  #if os(macOS)
+    private func showSignedOutAlertmacOS() {
+      let alert = NSAlert()
+      alert.messageText = "Your Firezone session has ended"
+      alert.informativeText = "Please sign in again to reconnect"
+      NSApp.activate(ignoringOtherApps: true)
+      alert.runModal()
```

Comment:
Is it possible to change the buttons on this alert to:

`Dismiss | Sign in` so that the user can just click Sign in to fix it without having to click in the menubar?

Also, are the macOS notifications running in Notification center or are they separate? Do they auto-dismiss or stay visible until user action?

Reply:
> `Dismiss | Sign in` so that the user can just click Sign in to fix it without having to click in the menubar?

Makes sense, will do that.

> Also, are the macOS notifications running in Notification center or are they separate? Do they auto-dismiss or stay visible until user action?

iOS-like notifications are available in macOS as well, they behave similar to iOS (by default they auto-disappear but the user can make them stay in OS Settings). However, they are easy to miss (because they are small and in a corner), so we're using an in-app alert instead.

Reply:
Gotcha. I'm wondering if it makes sense to use the Notification Center notification for macOS as well so that user's can turn it off if needed and so that it respects the Do Not Disturb settings by the user.

Otherwise it might be annoying to have that alert pop up in the middle of a presentation or something with no way to deactivate or see how much longer you have left in your session.

If they miss and notice the resources can't be accessed they'll probably figure out to click on the Firezone in menubar.

What do you think?